bentoml/yatai/client/bento_repository_api.py

```diff
@@ -344,62 +346,62 @@ def list(
+        Delete bentos that matches the specified criteria
 
         Args:
-            bento: a BentoService identifier in the format of NAME:VERSION
-
+            all: boolean, Set True to delete all BentoService
+            labels: string
+            bento_name: string
+            bento_version: string
+            confirm_delete: boolean
         Example:
         >>>
-        >>> yatai_client = get_yatai_client()
-        >>> yatai_client.repository.delete('my_service:version')
-        """
-        track('py-api-delete')
-        if ':' not in bento:
-            raise BentoMLException(
-                'BentoService name or version is missing. Please provide in the '
-                'format of name:version'
-            )
-        name, version = bento.split(':')
-        result = self.yatai_service.DangerouslyDeleteBento(
-            DangerouslyDeleteBentoRequest(bento_name=name, bento_version=version)
-        )
-        if result.status.status_code != yatai_proto.status_pb2.Status.OK:
-            error_code, error_message = status_pb_to_error_code_and_message(
-                result.status
-            )
-            raise BentoMLException(
-                f'Failed to delete Bento {bento} {error_code}:{error_message}'
-            )
-
-    def prune(self, bento_name=None, labels=None):
-        """
-        Delete all BentoServices that matches the specified criteria
-
-        Args:
-            bento_name: optional
-            labels: optional
-
-        Example:
-
         >>> yatai_client = get_yatai_client()
         >>> # Delete all bento services
-        >>> yatai_client.repository.prune()
-        >>> # Delete bento services that matches with the label `ci=failed`
-        >>> yatai_client.repository.prune(labels='ci=failed')
+        >>> yatai_client.repository.delete(all=True)
+        >>> # Delete bento service with name is `IrisClassifier` and version `0.1.0`
+        >>> yatai_client.repository.delete(
+        >>>     bento_name='IrisClassifier', bento_version='0.1.0'
+        >>> )
+        >>> # Delete all bento services with name 'MyService`
+        >>> yatai_client.repository.delete(bento_name='MyService')
+        >>> # Delete all bento services with labels match `ci=failed` and `cohort=20`
+        >>> yatai_client.repository.delete(labels='ci=failed, cohort=20')
         """
-        track('py-api-prune')
-        list_bentos_result = self.list(bento_name=bento_name, labels=labels,)
-        if list_bentos_result.status.status_code != yatai_proto.status_pb2.Status.OK:
-            error_code, error_message = status_pb_to_error_code_and_message(
-                list_bentos_result.status
+        track('py-api-delete')
+        bento_delete_list = []
+        if all is True:
+            bento_delete_list = self.list()
+        else:
+            if bento_name is not None and bento_version is not None:
+                bento_delete_list.append(self.get(f'{bento_name}:{bento_version}'))
```

Comment:
logging: `Deleting saved bundle f'{bento_name}:{bento_version}'`

Reply:
I think you forgot the case of `delete(name="my_service")` here, which deletes all versions of the BentoService with that same name.

Pls add a test to cover it.

Reply:
Similarily add a logging message here saying `Deleting all {len(...)} versions of saved bundles with name '{name}'`

Reply:
> I think you forgot the case of `delete(name="my_service")` here, which deletes all versions of the BentoService with that same name.
> 
> Pls add a test to cover it.

deleted all versions of the bento service with the same name is covered with `self.list()`. We have a test for this case.  I will add a better logging message for all of the

Reply:
see my other comment below, let's make that case a separate branch